### PR TITLE
Fix show_chunks on hypertable by hash

### PR DESCRIPTION
**Pull request by fabriziomello**
Using the new hypertable API creating closed dimentions when partitioning by hash was leading to a segfault on show_chunks.

Fixed it by checking for the closed primary dimension and erroring out in case of using parameters `older_than` and `newer_than`.

Disable-check: force-changelog-file

### CODE REVIEW

File: src/chunk.c [hunk 1]
```diff
@@ -2178,6 +2189,9 @@ get_chunks_in_time_range(Hypertable *ht, int64 older_than, int64 newer_than, Mem
 	end_strategy = (older_than == PG_INT64_MAX) ? InvalidStrategy : BTLessStrategyNumber;
 	time_dim = hyperspace_get_open_dimension(ht->space, 0);
 
+	if (time_dim == NULL)
+		time_dim = hyperspace_get_closed_dimension(ht->space, 0);
```
Comment (svenklemm) [inline src/chunk.c:2193]: aren't we replacing one assumption for another here? Should we add proper check in case this fails too?

Comment (fabriziomello) [reply src/chunk.c:2198]: added an `Ensure` to guarantee we have either open or close time dimension defined for the hypertable

File: test/expected/chunk_utils.out [hunk 2]
```diff
@@ -1601,3 +1601,58 @@ SELECT * FROM test_view_part_few WHERE ts BETWEEN '2024-01-04 00:00:00+00'AND '2
          15 | Thu Jan 04 21:55:12 2024 PST
 (14 rows)
 
+-- Hypertable by hash partitioning
+CREATE TABLE test_by_hash (id BIGINT, value FLOAT8);
+SELECT create_hypertable('test_by_hash', by_hash('id', 10));
+ create_hypertable
```
Comment (nikkhils) [inline test/expected/chunk_utils.out:1607]: @fabriziomello with my PR https://github.com/timescale/timescaledb/pull/6993 we won't allow this.

Comment (fabriziomello) [reply test/expected/chunk_utils.out]: Fixed!

**Review by svenklemm — APPROVED** [review]

File: src/chunk.c [hunk 3]
```diff
@@ -2178,6 +2189,14 @@ get_chunks_in_time_range(Hypertable *ht, int64 older_than, int64 newer_than, Mem
...
+
+	Ensure(time_dim != NULL,
+		   "time dimension not found for hypertable \"%s\".\"%s\"",
```
Comment (nikkhils) [inline src/chunk.c:2196]: @fabriziomello should the message be: "partitioning dimesion not found for hypertable"?

Comment (fabriziomello) [reply src/chunk.c]: Fixed!

**Review by nikkhils — APPROVED** [review]
LGTM!

**Pull request merged**